Lib/test/test_readline.py

```diff
@@ -349,6 +350,33 @@ def test_history_size(self):
             self.assertEqual(len(lines), history_size)
             self.assertEqual(lines[-1].strip(), b"last input")
 
+    @requires_working_threading()
+    @requires_gil_enabled()
+    def test_gh123321_threadsafe(self):
+        """gh-123321: readline should be thread-safe and not crash"""
+        script = r"""
+import threading
+
+def func():
+    input()
+
+thread1 = threading.Thread(target=func)
+thread2 = threading.Thread(target=func)
+thread1.start()
+thread2.start()
+thread1.join(3)
+thread2.join(3)
```

Comment:
Please use `from test.support.threading_helper import join_thread` and then use `join_thread()` to join the thread: it raises an exception if the thread is still alive, and it avoids an hardcoded timeout. So you can remove the is_alive() check.

Reply:
Good to know it exists 😄